src/target/riscv/riscv.c

```diff
@@ -743,12 +869,21 @@ static int try_setup_chained_match_triggers(struct target *target,
 	for (unsigned int idx = 0;
 			find_next_free_trigger(target, trigger_type, true, &idx) == ERROR_OK;
 			++idx) {
-		ret = set_trigger(target, idx, t1.tdata1, t1.tdata2,
-				t1.tdata1_ignore_mask);
-		if (ret != ERROR_OK)
+		ret = try_use_trigger_and_cache_result(target, idx, t1.tdata1, t1.tdata2,
+			t1.tdata1_ignore_mask);
+
+		switch (ret) {
+		case ERROR_OK:
+			break;
+		case ERROR_TARGET_RESOURCE_NOT_AVAILABLE:
 			continue;
-		ret = set_trigger(target, idx + 1, t2.tdata1, t2.tdata2,
-				t2.tdata1_ignore_mask);
+		default:
+			return ret;
+		}
```

Comment:
This is correct, but not idiomatic.

I'd go with:
```
if (ret == ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
   continue;
else if (ret != ERROR_OK)
   return ret;
```

Reply:
Addressed.

Reply:
Did you forget to push this?

Reply:
Bump.

Reply:
Addressed.